packages/dds/tree/src/simple-tree/core/allowedTypes.ts

```diff
@@ -206,47 +435,31 @@ export type ImplicitAllowedTypes = AllowedTypes | TreeNodeSchema;
 /**
  * Types of {@link TreeNode|TreeNodes} or {@link TreeLeafValue|TreeLeafValues} allowed at a location in a tree with
  * additional metadata associated with the location they're allowed at.
+ * @remarks
+ * Since AllowedTypesFull implements `AllowedTypes`, this type is not really necessary and ImplicitAllowedTypes could be used instead.
  * @alpha
  * @input
  */
-export type ImplicitAnnotatedAllowedTypes =
-	| TreeNodeSchema
-	| AnnotatedAllowedType
-	| AnnotatedAllowedTypes
-	| readonly (AnnotatedAllowedType | LazyItem<TreeNodeSchema>)[];
+export type ImplicitAnnotatedAllowedTypes = AllowedTypesFull | ImplicitAllowedTypes;
 
 /**
- * Returns an {@link ImplicitAllowedTypes} that is equivalent to the input without annotations.
+ * Removes annotations from a list of allowed types that may contain annotations.
  * @system @alpha
  */
-export type UnannotateImplicitAllowedTypes<T extends ImplicitAnnotatedAllowedTypes> =
```

Comment:
This is the logic I was unable to make a duplicate copy of for recursive types which worked. Deleting it is a nice solution to that probem.